mods/page_save.js

```diff
@@ -154,7 +172,8 @@ PageSave.prototype.saveWikitext = function(restbase, req) {
         return restbase.post({
             uri: new URI([rp.domain, 'sys', 'action', 'edit']),
             headers: {
-                cookie: req.headers.cookie
+                cookie: req.headers.cookie,
+                'x-forwarded-for': req.headers['x-forwarded-for']
```

Comment:
Hm. If the API doesn't do anything sensible when XFF is missing then we should ask ourselves what's the point of this PR, shouldn't we?

You are right, though, that we should supply an agreed-upon value in case XFF is missing, so that it can be white-listed on the API servers and that we don't hit any limits or restrictions imposed on individual IPs.